letsencrypt-auto-source/build.py

```diff
@@ -21,14 +21,17 @@ def build(version=None, requirements=None):
     :arg version: The version to attach to the script. Default: the version of
         the certbot package
     :arg requirements: The contents of the requirements file to embed. Default:
-        contents of letsencrypt-auto-requirements.txt
+        contents of dependency-requirements.txt, letsencrypt-requirements.txt,
+        and certbot-requirements.txt
 
     """
     special_replacements = {
         'LE_AUTO_VERSION': version or certbot_version(DIR)
     }
     if requirements:
-        special_replacements['letsencrypt-auto-requirements.txt'] = requirements
```

Comment:
If the caller provides the optional `requirements` value it is included in `letsencrypt-auto` rather than our normal requirements. These lines cause the templates to be expanded using the contents of the dictionary rather than the normal file. I assign two values to an empty string so only the caller's requirements are used rather than mixing in ours.

This functionality is used in our [tests](https://github.com/certbot/certbot/blob/master/letsencrypt-auto-source/tests/auto_test.py#L255).